### ISSUE TITLE
Password masking show/hide should be button not link

### ISSUE THREAD

**Issue by patrickcate** [uswds/uswds#4592]
**Is your feature request related to a problem? Please describe.**
On most of the authentication templates there is a password masking show/hide functionality shown (example: https://designsystem.digital.gov/templates/authentication-pages/sign-in/)

The toggle for this is a `a` tag with the appropriate aria attributes and JS to enable the toggling. Since this interactive control is not used for navigation it should instead be implemented with a `button` element.

**Describe the solution you'd like**
Replace the `a` tag with a `button` and style it as a link.

**Comment by shirshak55** [comment]
Additionally, it would be better if they use eye icons.
